mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp

```diff
@@ -990,9 +990,12 @@ tileAndFuseFirstExtractUse(RewriterBase &rewriter, Diagnostic &diag,
   // from the bbArg instead. This allows to reuse the output tensor (instead of
   // creating a new one) of the container when both producer and container write
   // to the same output.
+  bool cloned = false;
   if (LoopLikeOpInterface containerLoop =
-          dyn_cast<LoopLikeOpInterface>(sliceOpToTile->getParentOp())) {
+          dyn_cast<LoopLikeOpInterface>(sliceOpToTile->getParentOp());
+      containerLoop && dyn_cast<DestinationStyleOpInterface>(producerOp)) {
```

Comment:
Normally, one should use `isa` rather than `dyn_cast` when the result of the cast is unused. However, having an `isa` followed by a `cast` (below) is an anti-pattern.  You can hoist `dyn_cast` to happen before the loop and just check the value for not being nullptr in the condition and use it below without the cast.